### PR TITLE
Apicall exports all functions and UI indicates a mistake

### PR DESCRIPTION
**Pull request by LucileTech**
## I'm submitting a ...

- feature request
Needs Error handling for chat api call #59

## What is the current behavior?
Code logic moved and UI updated to indicate a mistake

## What is the expected behavior?
Create a New File for API Calls: Create a new file (e.g., api.js or api.ts) and move the API logic there. This file will export functions to fetch tasks and send messages.
Graceful Error Handling: Ensure error responses are handled gracefully in the new file. Instead of just printing and returning errors, provide meaningful feedback to the user, such as showing a notification or updating the UI to indicate a mistake.

## What is the motivation / use case for changing the behavior?
"Each React component should ideally have a single responsibility, and handling API responses in the component layer goes against this principle."

### CODE REVIEW

**Comment by huyfififi** [comment]
will review by tmr

File: frontend/README.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
 
+### Set Up Environment Variables
+
+Before starting the development server, make sure to set the `NEXT_PUBLIC_API_BASE_URL` environment variable in a `.env` file:
+
+````bash
```
Comment (huyfififi) [inline frontend/README.md:7]: I think using three backticks is more appropriate in Markdown.
https://www.markdownguide.org/extended-syntax/#fenced-code-blocks

Comment (huyfififi) [reply frontend/README.md:7]: Same goes for line 22

File: frontend/README.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
 
+### Set Up Environment Variables
+
+Before starting the development server, make sure to set the `NEXT_PUBLIC_API_BASE_URL` environment variable in a `.env` file:
```
Comment (huyfififi) [inline frontend/README.md:5]: 👍

**Review by huyfififi — APPROVED** [review]
👍

**Comment by LucileTech** [comment]
Thank you for your review @huyfififi !

**Pull request merged**